pages/components/cards.js

```diff
@@ -209,6 +224,24 @@ export default function Cards({ posts, setPosts }) {
                   <button type="submit">Submit</button>
                 </form>
               )}
+              {post.isEditing && (
```

Comment:
the information whether a post is being edited or not should not be part of the post itself, but a separate state